datahub/dsr.py

```diff
@@ -40,24 +35,23 @@ class Config:
         allow_population_by_field_name = True
 
 
-def validate_dsr_sizes(data: dict[str, Any]) -> list[str]:  # type: ignore[misc]
+def validate_dsr_arrays(data: dict[str, Any]) -> list[str]:  # type: ignore[misc]
     """Validate the sizes of the arrays in the DSR data.
 
     Args:
         data: The dictionary representation of the DSR Data. The keys are field aliases.
+            It is generated with the data.dict(by_alias=True) where data is a DSRModel.
 
     Returns:
         An empty list if there are no issues. A list of the failing fields if there are.
     """
     aliases = []
     for field in list(DSRModel.__fields__.values()):
-        if field.annotation == list[list[float]]:
-            rows = len(data[field.alias])
-            cols = len(data[field.alias][0])
-            if (rows, cols) != field.field_info.extra["size"]:
-                aliases.append(field.alias)
-        elif field.annotation == list[float]:
-            rows = len(data[field.alias])
-            if (rows,) != field.field_info.extra["size"]:
+        if field.annotation == list:
```

Comment:
I don't really like having to use a private attribute of `DSRModel` - they are not supposed to be used externally. What about the following approach:

```python
for alias, value in data.items():
     if not isinstance(value, list):
          continue

     array = ...
     if ... :
          aliases.append(alias)

return aliases
```

Reply:
I've realised why this is not valid - you need also the expected `size`. In that case, I would hide the use of the restricted member of `DSRModel` behind a public interface, something like:

```python
@classmethod
def fields(cls):
    return list(cls.__fields__.values())
```

And then:
```python
for alias in DSRModel.fields():
     if field.annotation != list:
          continue

     array = ...
     if ... :
          aliases.append(alias)

return aliases
```

Reply:
Agreed. There is an inbuilt `schema()` method that I found that can get the data I need here